src/main/java/de/sakpaas/backend/service/SearchMappingService.java

```diff
@@ -39,51 +44,106 @@ public SearchMappingService(RestTemplate restTemplate,
                               MeterRegistry meterRegistry) {
     this.restTemplate = restTemplate;
     this.meterRegistry = meterRegistry;
+  }
 
+  /**
+   * Searches in the Nominatim Microservice for the given query.
+   *
+   * @param query The search parameter
+   * @return The coordinates of the search request
+   * @throws IndexOutOfBoundsException Iff the the request returned nothing
+   */
+  public CoordinateDetails search(Set<String> query) throws IndexOutOfBoundsException {
+    buildUrl(query);
+    encodeUrl();
+    return returnCoordinates();
+  }
 
+  /**
+   * Searches in the Nominatim Microservice for the given query.
+   *
+   * @param query The search parameter
+   * @return The coordinates of the search request
+   * @throws IndexOutOfBoundsException Iff the the request returned nothing
+   */
+  public CoordinateDetails search(Set<String> query, CoordinateDetails coordinateDetails)
+      throws IndexOutOfBoundsException {
+    buildUrl(query, coordinateDetails);
+    encodeUrl();
+    return returnCoordinates();
   }
 
   /**
-   * Searches in the Nominatim Microservice for the given key.
+   * Encodes the nominatim request query.
+   * To be implemented in TINF-272.
    *
-   * @param key The search parameter. Multiple words are separated with %20.
-   * @return The list of Locations in our database
+   * @param query The search parameter
    */
-  public CoordinateDetails search(String key) {
-    String url = this.searchApiUrl + "/search/" + key + "?format=json";
+  private void buildUrl(Set<String> query) {
+    String queryString = String.join(",", query);
+    this.url = this.searchApiUrl + "/search/" + queryString + "?format=json&limit=1";
```

Comment:
Du könntest die URL auch schon direkt hier korrekt bauen:
```
String uri = UriComponentsBuilder.fromHttpUrl("http://nominatim.openstreetmap.org/search")
        .queryParam("q", query)
        .queryParam("limit", 1)
        .toUriString();
```
Den Code hatten wir schon in TINF-273 vorgeschlagen, aber mir ist aufgefallen dass das Ticket nur verlinkt war und der Code nicht auch nach TINF-272 kopiert wurde.